### DISCUSSION TITLE
Package "esbuild-darwin-arm64" could not be found, and is needed by esbuild.

### DISCUSSION THREAD

**Discussion by cole-robertson** [fwouts/previewjs#49] [Q&A]
Hey, this package looks awesome. I think most of us have been waiting to preview React components right in our IDE for some time. 

I do have a problem with installation of the Vs Code extension though. When trying to click to "Open preview" I get error: 'Package "esbuild-darwin-arm64" could not be found, and is needed by esbuild' in Vscode. 

I am on an M1 Mac.

Have you encountered this before or know any way to correct it? Thank you so much.

**Comment by cole-robertson** [comment]
https://www.cypress.io/blog/2021/01/20/running-cypress-on-the-apple-m1-silicon-arm-architecture-using-rosetta-2/

Found this link that I may give a go later.

**Comment by fwouts** [comment]
Hey @cole-robertson, I develop React Preview on an M1 Mac and haven't had that issue before!

What version of Node do you use? (`node --version`)

It's also possible that this is due to a bug that got fixed just a few hours ago in esbuild: https://github.com/evanw/esbuild/releases/tag/v0.13.4. You may need to delete `~/.vscode/extensions/zenclabs.reactpreview-0.7.1` before reinstalling so that React Preview can reinstall dependencies in a fresh state.

**Reply by cole-robertson** [reply]
I am using node version 14.17.0. I tried to `rm -rf ~/.vscode/extensions/zenclabs.reactpreview-0.7.1` and reinstall, but it still seems to fail with the same error when trying to open a component in preview.

You aren't running your vs code in Rosetta are you?

**Reply by fwouts** [reply]
Interesting, here's my VS Code info for comparison:
```
Version: 1.60.2
Commit: 7f6ab5485bbc008386c4386d08766667e155244e
Date: 2021-09-22T11:59:08.826Z (1 wk ago)
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin arm64 20.6.0
```

The extension runs `npm install` on start, which might be using a different version of Node. I suspect in your case, it's running an x86-64 version of Node, which means is incompatible with the Node version used by VS Code. Would that make sense?

You may be able to fix it by running `npm install esbuild --no-optional` from `.vscode/extensions/zenclabs.reactpreview-0.7.1/dist/server`. Let me know how you go!

**Reply by cole-robertson** [reply]
Yes, looks the same as mine besides I am using the "universal" version of vscode. 
```
Version: 1.60.2 (Universal)
Commit: 7f6ab5485bbc008386c4386d08766667e155244e
Date: 2021-09-22T11:59:27.195Z (1 wk ago)
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin arm64 20.6.0
```

I tried to run the `--no-optional` command in the extension folder, but I am still having the same problem. Are you running the M1 native version of Vscode?

**Reply by cole-robertson** [reply]
Ah great I seemed to fix it by clearing the node_modules manually `rm -rf node_modules` and then running `npm install esbuild` without the "--no-optional" flag.

**Reply by fwouts** [reply]
Awesome! I'll see if I can reproduce and introduce a fix on my end :)

**Reply by benedelstein** [reply]
I had the same issue and cole's solution worked for me as well.

**Reply by alex-oos** [reply]
> Ah great I seemed to fix it by clearing the node_modules manually `rm -rf node_modules` and then running `npm install esbuild` without the "--no-optional" flag.

This method is particularly effective, thank you ~

**Reply by hofstede-matheus** [reply]
Removing `node_modules` did not work for me, but forcing no caching in npm did `npm install --force`
